### ISSUE TITLE
configurable "filtered" view

### ISSUE THREAD

**Issue by michalskrivanek** [oVirt/ovirt-web-ui#317]
We changed the behavior of the old portal to show all the VMs user has access to (all returned by REST API). But this is not a great behavior for everyone. Sometimes people have to use an admin permission for otherwise regular user to allow some specific operation in webadmin. But it doesn't mean user should see/access all the VMs, at least not in an obvious way (since he/she can't run them anyway)
Let's ake the behavior configurable on the server side, something like "Display only VMs with UserVM permission" option

**Comment by jelkosz** [comment]
TL;DR: There is a complaint that now if a user has any admin permission, ovirt-web-ui always sends Filter:False header and the user gets much more results than he used to (since previously the user portal always sent Filter: True which was btw also confusing and complained about). This are the options how to solve it:

After some offline discussions with @michalskrivanek and @mareklibra we agree that it should be controlled by an engine-config property but the actual layer using it is not decided.

We have this two options:

1: filter on frontend, which would go like this:
  - have an engine-config property named like: AlwaysFilterResultsForWebUi
  - ovirt-web-ui would read this property
    - If the particular user has some admin permission, than:
      - if the value of AlwaysFilterResultsForWebUi == true, send Filter: True, else send Filter: False
      - if the  AlwaysFilterResultsForWebUi is not present in engine-config, fallback to send Filter: False
    - if the user has no admin permission, always send Filter: True

2: filter on server
  - have an engine-config property named like: AlwaysFilterResults
  - now, on the REST API layer, if this property is set to true, ignore the Filter header and always act like Filter: True was set

Pros and Cons:
1 (frontend filter): 
  - pros:
    - does not affect existing clients
    - easy to implement
  - cons:
    - it is not secure. But it is a question if it should be secure or rather just user friendly and if someone uses e.g. curl than will get the results is a problem

2 (server filter): 
  - pros:
    - secure (can not be worked around by e.g. curl)
    - also easy to implement :)
  - cons
    - affects existing clients

@jhernand what do you think? Any preference from your side?

**Comment by jelkosz** [comment]
OK, lets go by an engine-config option. It is easy to use for the users as everything is configured by it. And when https://gerrit.ovirt.org/#/c/81929/ will be merged, we can use it here.

Until done, setting this bug as blocked.

**Comment by jelkosz** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1441501

ok, done, we have vdc_options available in API. Lets unblock this.
